server/konnectors/linkedin.coffee

```diff
@@ -0,0 +1,340 @@
+
+    listUrls
+
+# Currently there isn't an address parser so all the address is set in the
+# locality fields. The Linkedin API allow us to define some fields precisely but
+# that can lead to a duplicat so only the country is specify in the right fields
+# and we should wait an address parser before use them
+getAddresses = (data) ->
+    listAddresses = []
+
+    if data.location?
+        segmentAddress = data.location.split(', ')
+            .reverse()
+        country = segmentAddress[0] || ''
+        #region = segmentAddress[1] || ''
+        #locality = segmentAddress[2] || ''
+
+    data.addresses?.forEach (address) =>
+        addressArray = ContactHelper.adrStringToArray address.raw
+        addressArray[6] = country
+
+        listAddresses.push
+            name: 'adr'
+            value: addressArray
+            type: 'main'
+
+    listAddresses
+
+
+
+
+# Save and/or merge the imported contacts with the existing
+saveContact  = (linkContact, entries) ->
+    linkAccount = ContactHelper.getAccount linkContact, ACCOUNT_TYPE, entries.accountName
+
+
+    urlImage = linkContact.imageUrl
+    delete linkContact.imageUrl
+
+    endSavePicture = (err, updatedContact) ->
+        if err?
+            log.error "An error occured while creating or updating the " + \
+                      "contact."
+            log.raw err
+            done err
+
+
+        if urlImage?
+            opts = url.parse(urlImage)
+            https.get opts, (stream) ->
+                stream.on 'error', (err) -> log.error err
+
+                updatedContact.attachFile stream, {name: 'picture'}, (err)->
+                    if err
+                        log.error "picture #{err}"
+                    else
+                        log.debug "picture ok"
+
+
+    updateContact = (fromCozy, fromLinkedin) ->
+        CompareContacts.mergeContacts fromCozy, fromLinkedin
+        fromCozy.save endSavePicture
+
+
+    if not linkAccount?
+        throw new Error "Contact Account not created:"
+
+    if entries.ofAccountByIds[linkAccount.id]?
+        cozyContact = entries.ofAccountByIds[linkAccount.id]
+        cozyAccount = cozyContact.getAccount ACCOUNT_TYPE, entries.accountName
+        if ContactHelper.intrinsicRev(linkContact) isnt ContactHelper.intrinsicRev(cozyContact)
+            log.info "Update #{cozyContact.fn} from LinkedIn"
+            log.debug "Update #{linkContact.fn} from LinkedIn"
+            updateContact cozyContact, linkContact
+
+        else # Already uptodate, nothing to do.
+            log.info "LinkedIn contact #{cozyContact.fn} already synced and uptodate"
+            log.debug "LContact #{linkContact.n}already synced and uptodate"
+    else
+        for cozyContact in entries.cozyContacts
+            if CompareContacts.isSamePerson cozyContact, linkContact
+                log.info "Link #{cozyContact.fn} to linkedin account"
+                updateContact cozyContact, linkContact
+                return
```

Comment:
This return is very ugly. Moreover, comparing each linkedin contacts with all Cozy contacts look very consuming. I think we should be more laxist about Contacts existence checking and only base it on name comparison. There is a duplicate feature in the Contact application that can handle duplicates later.